src/highlight/HighlightList.tsx

```diff
@@ -1,20 +1,48 @@
 import React from 'react';
 import classNames from 'classnames';
-import HighlightAnnotation from './HighlightAnnotation';
+import HighlightCanvas from './HighlightCanvas';
+import HighlightSvg from './HighlightSvg';
+import HighlightTarget from './HighlightTarget';
 import useOutsideEvent from '../common/useOutsideEvent';
-import { AnnotationHighlight } from '../@types';
-
+import { AnnotationHighlight, Rect } from '../@types';
+import { checkValue } from '../utils/util';
 import './HighlightList.scss';
 
 export type Props = {
-    annotations: AnnotationHighlight[];
+    activeId?: string | null;
+    annotations: Pick<AnnotationHighlight, 'id' | 'target'>[];
     className?: string;
     onSelect?: (annotationId: string | null) => void;
 };
 
-export function HighlightList({ annotations, className, onSelect }: Props): JSX.Element {
+export function isValidHighlight({ target }: Pick<AnnotationHighlight, 'target'>): boolean {
+    const { shapes = [] } = target;
+
+    return shapes.reduce((isValid: boolean, rect: Rect) => {
+        const { height, width, x, y } = rect;
+        return isValid && checkValue(height) && checkValue(width) && checkValue(x) && checkValue(y);
```

Comment:
Do we want to reject only the invalid shape rather than the entire annotation? Probably a question for product.

Reply:
Design confirms that we ignore the whole highlight